docs/pages/core-concepts.mdx

```diff
@@ -0,0 +1,219 @@
+**roles**, **local users**, and **authentication connectors**
+
+Read more in our [Resource Reference](./reference/resources.mdx).
+
+## Machine ID
+
+Machine ID is a service that automatically provisions credentials for **bot
+users**, which can use these credentials to connect automatically to resources
+in your infrastructure. Machine ID runs via the `tbot` binary.
+
+Read more in our [Machine ID guide](./machine-id/introduction.mdx).
+
+## Role
+
+A **role** is a **configuration resource** that grants **Teleport users**
+privileges within a cluster. Teleport's role-based access control (RBAC) is
+restrictive by default, and a user needs explicit permissions before they can
+access a resource or perform management tasks on a cluster.
+
+Read our [guide to Teleport roles](./access-controls/guides/role-templates.mdx).
+
+## Teleport Auth Service
+
+The **Teleport Auth Service** manages **certificate authorities**, **local
+users**, and **configuration resources** within a **Teleport cluster**. It uses
+storage backends and—if configured—a hardware security module to store the
+certificate authorities' private keys. The Auth Service issues certificates to
+clients and maintains an audit log.
+
+<Figure
+  align="center"
+  bordered="false"
+  caption="Auth service"
+>
+
+![Auth service](../img/architecture/auth.png)
+
+</Figure>
+
+The Auth Service is the only service that has to be connected to a backend. All
+other services are stateless and interact with the Auth Service via a gRPC API.
+**Teleport Services** can connect to the Auth Service either directly or via a
+reverse tunnel established by the **Proxy Service**. This means that you can run
+Teleport Services and the Auth Service in a private network, with only the Proxy
+Service's HTTPS port exposed to the internet.
+
+You can run multiple Auth Service instances in a cluster for high availability.
+
+Read our guides to how [authorization](./architecture/authorization.mdx) and
+[authentication](./architecture/authentication.mdx) work in Teleport.
+
+## Teleport Cloud
+
+**Teleport Cloud** is a managed deployment of the **Teleport Control Plane**.
+Our team at Teleport handles all tasks related to running the **Auth Service**
+and **Proxy Service**, including upgrades and certificate management. Each
+customer account, known as a **Teleport Cloud tenant**, has its own subdomain of
+`.teleport.sh`, e.g., `mytenant.teleport.sh`.
+
+Read more in our [Teleport Cloud
+guide](./deploy-a-cluster/teleport-cloud/introduction.mdx).
+
+## Teleport Cluster
+
+The key concept of Teleport's architecture is the **cluster**. A Teleport
+cluster consists of the **Teleport Auth Service** and **Teleport Proxy Service**,
+plus several **Teleport Services** that manage traffic to resources within your
+infrastructure, such as Kubernetes clusters and Windows desktops. A Teleport
+cluster also includes **Machine ID** and **Teleport Discover**.
```

Comment:
> A Teleport cluster also includes **Machine ID** and **Teleport Discover**

This part didn't make sense to me. I would clarify that Machine ID and Teleport Discover are both services you can run in a cluster.

Reply:
Teleport Discover isn't really a service that you run in the cluster though - is it?

I'm not sure we should even mention discover here. That feels more like an internal name to me. It's not used anywhere in the UI (or docs, to my knowledge)

Reply:
[This PR](https://github.com/gravitational/teleport/pull/16279) introduces the "Teleport Discovery Service", which maybe we could mention instead of Teleport Discover?